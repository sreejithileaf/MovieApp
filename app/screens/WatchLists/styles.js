/**
 * Created on July 17, 2022
 * Dashboard Screen - About Us Screen Styles
 */

import { StyleSheet } from "react-native";
import Constants from "../../config/constants";

const styles = StyleSheet.create({
  safeContainer: {
    flex: 1,
    backgroundColor: Constants.APP_TAB_COLOR,
  },
  headingTextStyle: {
    alignItems: "flex-start",
    fontSize: 34,
    fontFamily: Constants.Fonts.MEDIUM,
    color: Constants.APP_WHITE_COLOR,
    marginTop: 50,
    marginStart: 20,
    marginEnd: 20,
  },

  subHeadingOneTextStyle: {
    alignItems: "flex-start",
    fontSize: 16,
    fontFamily: Constants.Fonts.MEDIUM,
    color: Constants.APP_WHITE_COLOR,
    marginTop: 22,
    marginStart: 20,
    marginEnd: 20,
  },

  backButtonContainer: {
    position: "absolute",
    bottom: 10,
    right: 30,
    flexDirection: "row",
    justifyContent: "center",
    width: 105,
    height: 42,
    backgroundColor: Constants.APP_GENRES_COLOR,
    borderRadius: 16,
  },

  movieTextStyle: {
    color: Constants.APP_WHITE_COLOR,
    fontFamily: Constants.Fonts.REGULAR,
    fontWeight: "500",
  },

  watchListStyle: {
    flex: 1,
    marginTop: 30,
    marginStart: 20,
    marginEnd: 20,
  },

  watchListContentStyle: {
    justifyContent: "center",
    alignContent: "center",
  },

  noMovieContainerStyle: {
    flex: 1,
    justifyContent: "center",
    alignSelf: "center",
  },

  noMovietextStyle: {
    fontSize: 14,
    fontFamily: Constants.Fonts.MEDIUM,
    color: Constants.APP_WHITE_COLOR,
  },

  backImageStyle: {
    width: 16,
    height: 14,
    alignSelf: "center",
    justifyContent: "center",
  },

  backTextStyle: {
    color: Constants.APP_LIGHTGREY_COLOR,
    marginStart: 10,
    fontSize: 14,
    alignSelf: "center",
  },
});

export default styles;
