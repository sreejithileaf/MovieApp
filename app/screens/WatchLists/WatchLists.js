/**
 * Created on July 17, 2022
 * WatchLists screen
 */

import {
  View,
  TouchableOpacity,
  Text,
  SafeAreaView,
  FlatList,
  Image,
  TouchableHighlight,
} from "react-native";

import styles from "./styles";
import React, { Component } from "react";
import Images from "../../config/images";
import Constants from "../../config/constants";
import MovieRow from "../../components/MovieRow";

export default class WatchLists extends Component {
  constructor(props) {
    super(props);
  }

  /*
   * Navigate to back
   */
  _didTapOnBackButton = () => {
    this.props.navigation.goBack();
  };

  /*
   * Navigates to Movie detail page
   */
  _onRowClick = (item) => {
    this.props.navigation.navigate("MovieDetails", { id: item.id, item });
  };

  render() {
    return (
      <SafeAreaView style={styles.safeContainer}>
        <Text style={styles.headingTextStyle}>Movies App</Text>
        <Text style={styles.subHeadingOneTextStyle}>Your Watch List</Text>

        {this.props.watchlistItems && this.props.watchlistItems.length ? (
          <FlatList
            style={styles.watchListStyle}
            data={this.props.watchlistItems}
            showsVerticalScrollIndicator={false}
            extraData={this.props.watchlistItems}
            renderItem={({ item }) => (
              <TouchableHighlight
                underlayColor={Constants.APP_TAB_COLOR}
                style={styles.watchListContentStyle}
                key={item.id}
                onPress={() => this._onRowClick(item)}
              >
                <MovieRow
                  isFromDashboard={false}
                  title={item.title}
                  releasedate={item.release_date}
                  image_url={Constants.APP_IMG_URL_SMALL + item.poster_path}
                  avgrating={item.vote_average}
                />
              </TouchableHighlight>
            )}
          />
        ) : (
          <View style={styles.noMovieContainerStyle}>
            <Text style={styles.noMovietextStyle}>
              {"No movies in the Wish List"}
            </Text>
          </View>
        )}

        <TouchableOpacity
          style={styles.backButtonContainer}
          onPress={() => this._didTapOnBackButton()}
        >
          <Image style={styles.backImageStyle} source={Images.back}></Image>
          <Text style={[styles.movieTextStyle, styles.backTextStyle]}>
            {"Back"}
          </Text>
        </TouchableOpacity>
      </SafeAreaView>
    );
  }
}
