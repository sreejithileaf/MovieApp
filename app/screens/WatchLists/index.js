/**
 * Created on July 16, 2022
 * DashboardScreen - DashboardScreen container
 */

import { connect } from "react-redux";
import React, { Component } from "react";
import WatchLists from "./WatchLists";

class WatchListsScreenContainer extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return <WatchLists {...this.props} />;
  }
}

function mapStateToProps(state) {
  return {
    isLoading: state.loadingReducer.isLoading,
    watchlistItems: state.watchlistReducer.watchlistItems,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    getMovieDetailsAction: (params, getMovieDetailsCallback) => {
      dispatch(
        getMovieDetails.getMovieDetails(params, getMovieDetailsCallback)
      );
    },
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(WatchListsScreenContainer);
