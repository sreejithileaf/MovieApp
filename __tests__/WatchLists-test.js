/**
 * @format
 */

import "react-native";
import React from "react";
import WatchLists from "../app/screens/WatchLists/WatchLists";
import Constants from "../app/config/constants";
import {
  render,
  container,
  screen,
  fireEvent,
} from "@testing-library/react-native";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

describe("Watch Lists Screen renders correctly", () => {
  const wrapper = renderer.create(<WatchLists />).toJSON();
  test("Snapshot ", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("It must have title", async () => {
    const { getByText } = render(<WatchLists />);
    getByText("Your Watch List", { exact: false });
  });
  it("Title style", async () => {
    const { getByText } = render(<WatchLists />);
    getByText("Your Watch List", { exact: false });
  });
});
