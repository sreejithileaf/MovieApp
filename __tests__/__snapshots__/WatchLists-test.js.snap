// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Watch Lists Screen renders correctly Snapshot  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#242A32",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "alignItems": "flex-start",
        "color": "#FFFFFF",
        "fontFamily": "Poppins-Regular",
        "fontSize": 34,
        "marginEnd": 20,
        "marginStart": 20,
        "marginTop": 50,
      }
    }
  >
    Movies App
  </Text>
  <Text
    style={
      Object {
        "alignItems": "flex-start",
        "color": "#FFFFFF",
        "fontFamily": "Poppins-Regular",
        "fontSize": 16,
        "marginEnd": 20,
        "marginStart": 20,
        "marginTop": 22,
      }
    }
  >
    Your Watch List
  </Text>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Poppins-Regular",
          "fontSize": 14,
        }
      }
    >
      No movies in the Wish List
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#3A3F47",
        "borderRadius": 16,
        "bottom": 10,
        "flexDirection": "row",
        "height": 42,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 30,
        "width": 105,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/images/back.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": 14,
          "justifyContent": "center",
          "width": 16,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Poppins-Regular",
            "fontWeight": "500",
          },
          Object {
            "alignSelf": "center",
            "color": "#EEEEEE",
            "fontSize": 14,
            "marginStart": 10,
          },
        ]
      }
    >
      Back
    </Text>
  </View>
</RCTSafeAreaView>
`;
